ivadomed/config/config_dataset.json

```diff
@@ -0,0 +1,480 @@
+                                "derivatives/labels/sub-02/ses-04/anat/sub-02_ses-04_mt-off_MTS_lesion-manual-rater1.nii"
+                            ]
+                        ]
+                    ],
+                    "expected_input": 2,
+                    "expected_gt": 1,
+                    "missing_files_handle": "skip",
+                    "excessive_files_handle": "use_first_and_warn",
+                    "path_data": "C:\\git\\ivadomed\\tmp"
+                }
+            ],
+            "test": [
+                {
+                    "type": "FILES",
+                    "subset_label": "TestFileDataSet1",
+                    "image_ground_truth": [
+                        [
+                            [
+                                "sub-03/ses-01/anat/sub-03_ses-01_flip-1_mt-off_MTS.nii",
+                                "sub-03/ses-01/anat/sub-03_ses-01_flip-1_mt-on_MTS.nii"
+                            ],
+                            [
+                                "derivatives/labels/sub-03/ses-01/anat/sub-03_ses-01_mt-off_MTS_lesion-manual-rater1.nii"
+                            ]
+                        ],
+                        [
+                            [
+                                "sub-03/ses-02/anat/sub-03_ses-02_flip-1_mt-off_MTS.nii",
+                                "sub-03/ses-02/anat/sub-03_ses-02_flip-1_mt-on_MTS.nii"
+                            ],
+                            [
+                                "derivatives/labels/sub-03/ses-02/anat/sub-03_ses-02_mt-off_MTS_lesion-manual-rater1.nii"
+                            ]
+                        ],
+                        [
+                            [
+                                "sub-03/ses-03/anat/sub-03_ses-03_flip-1_mt-off_MTS.nii",
+                                "sub-03/ses-03/anat/sub-03_ses-03_flip-1_mt-on_MTS.nii",
+                                "sub-03/ses-03/anat/sub-03_ses-03_flip-1_mt-on_MTS.nii"
+                            ],
+                            [
+                                "derivatives/labels/sub-03/ses-03/anat/sub-03_ses-03_mt-off_MTS_lesion-manual-rater1.nii"
+                            ]
+                        ],
+                        [
+                            [
+                                "sub-03/ses-04/anat/sub-03_ses-04_flip-1_mt-off_MTS.nii"
+                            ],
+                            [
+                                "derivatives/labels/sub-03/ses-04/anat/sub-03_ses-04_mt-off_MTS_lesion-manual-rater1.nii"
+                            ]
+                        ]
+                    ],
+                    "expected_input": 2,
+                    "expected_gt": 1,
+                    "missing_files_handle": "skip",
+                    "excessive_files_handle": "use_first_and_warn",
+                    "path_data": "C:\\git\\ivadomed\\tmp"
+                }
+            ],
+            "expected_input": 2,
+            "expected_gt": 1
```

Comment:
Sorry Yang, I know you explained it a few times but I'm still missing something regarding the different levels of `expected_input` and `expected_gt` 😅 

So the tensor that we feed the network is always the same size i.e. all pairs must have the same dimensions. If we have the expected numbers and do the "clean-up" (missing, excessive files) at the dataset level, why do we need them at higher level? Won't they always be the same? What happens if a higher level has a different expected numbers than the dataset level?

I just don't get how they could be different from dataset to dataset and from levels to levels. And if they have to always be the same, could we specify those expected numbers at an unique place in the config?

Reply:
Related https://github.com/ivadomed/ivadomed/issues/1087#issuecomment-1081039335 but still agree that the `expected_input` and `expected_gt` must be at the highest level to accommodate the above comment by Naga.

Reply:
Great question. It is a design choice to cover scenarios where there are conflicts... 

So... AllDataset Group Level: Highest. If not the same, hard pass any DatasetGroup within it. 
At Dataset Group Level: Higher but not the highest. If not the same at BIDSDataset/FIleDataset, hard pass any of these "dataset" regardless if they are in train/test/val etc. 
At DataSet level, it will skip any subjects that have less, usually trim (based on list order) of more than provided. So at the end, that dataset is guarnteed to have X for input and Y for GT, with validated file presence guarnteed.